bin/rename_fastq_headers_pre.py

```
@@ -0,0 +1,175 @@
+    # Illumina format pre-CASAVA 1.8
+    pattern1 = r"^\S+/[1,2]$"       # matches for example the following header 'example.1/1'
+    # Illumina format pre-CASAVA 1.8 with additional information
+    pattern2 = r"^\S+/[1,2]\s"      # matches for example the following header 'example.1/1 additionalInformation'
+    # Illumina format post-CASAVA 1.8
+    pattern3 = r"^\S+\s\S+$"        # matches for example the following header 'readID1 additionalTechnicalInformation'
+    # Illumina format post-CASAVA 1.8 with additional Information
+    pattern4 = r"^\S+\s\S+\s\S+"    # matches for example the following header 'readID1 additionalTechnicalInformation additionalInformation'
+    # Any other pattern without spaces
+    pattern5 = r"^\S+$"             # matches for example the following header 'readID1'
+    if bool(re.match(pattern1,read)):
+        read_stripped = read[:-2]
+        read_dict[read_stripped] = [read]
+        read_renamed = [read_stripped]
+    elif bool(re.match(pattern2,read)):
+        read_split = read.split(" ")[0]
+        read_stripped = read_split[:-2]
+        read_dict[read_stripped] = [read]
+        read_renamed = [read_stripped]
+    elif bool(re.match(pattern3,read)):
+        read_split = read.split(" ")[0]
+        read_dict[read_split] = [read]
+        read_renamed = [read_split]
+    elif bool(re.match(pattern4,read)):
+        read_split = read.split(" ")[0]
+        read_dict[read_split] = [read]
+        read_renamed = [read_split]
+    elif bool(re.match(pattern5,read)):
+            read_dict[read] = [read]
+            read_renamed = [read]
+    else:
+        sys.exit(f"The provided file, {filename}, contained reads with headers not supported by the pipeline.\n  Please use one of the formats:\n    example.1/1\n    example.1/1 additionalInformation\n    readID1 additionalTechnicalInformation\n    readID1 additionalTechnicalInformation additionalInformation\n    readID1\nAny other format is not supported.")
```

I see you've used `sys.exit` a few places in the Python-script. Does a call to `sys.exit("foo")` result in the pipeline failing loudly?  (I guess it should.)

---

Yes, as the files needed by the process to be emitted are not created -> the process will fail, not sure if the error is put to the console, but it would be in the `work` folder. Still, might be better to raise an error 👍